caravel/assets/visualizations/histogram.js

```diff
@@ -0,0 +1,146 @@
+// JS
+var d3 = require('d3')
+var px = window.px || require('../javascripts/modules/caravel.js')
+
+// CSS
+require('./histogram.css')
+
+function histogram(slice) {
+    
+    var div = d3.select(slice.selector)
+    
+    var _draw = function(data, noOfBins) {
+        
+        var margin = { top: 50, right: 10, bottom: 20, left: 50 },
+            navBarHeight = 36,
+            navBarTitleSize = navBarHeight / 3,
+            navBarBuffer = 10,
+            width = slice.width() - margin.left - margin.right,
+            height = slice.height() - margin.top - margin.bottom - navBarHeight - navBarBuffer,
+            formatNumber = d3.format(",.0f"), 
+            formatTicks = d3.format(",.00f");
+
+        var bins = d3.layout.histogram().bins(noOfBins)(data),
+            x = d3.scale.ordinal(),
+            y = d3.scale.linear(),
+            xAxis = d3.svg.axis().scale(x).orient("bottom").ticks(noOfBins).tickFormat(formatTicks),
+            yAxis = d3.svg.axis().scale(y).orient("left").ticks(noOfBins*3);
+ 
+        x.domain(bins.map(function(d) { return d.x;}))
+         .rangeRoundBands([0, width], .1); 
+        
+        y.domain([0, d3.max(bins, function(d) { return d.y;})])
+         .range([height, 0]);
+
+        var svg = div.selectAll("svg").data([bins]).enter().append("svg");
+        
+        svg.append("rect")
+           .attr("width", "100%")
+           .attr("height", "100%")
+           .attr("fill", "#f6f6f6");
+
+        var gEnter = svg
+            .append("g")
+            .attr("transform", "translate(" + margin.left + "," + margin.top + ")");
+        
+        gEnter.append("g").attr("class", "bars");
+        gEnter.append("g").attr("class", "x axis");
+
+        svg.attr("width", slice.width())
+           .attr("height", slice.height());
+        
+        var g = svg.select("g")
+            .attr("transform", "translate(" + margin.left + "," + margin.top + ")");
+        
+        function colors(n) {         
+            var color_list = [
```

Comment:
The other modules usually pick a color from px.color.category21 from javascripts/modules/caravel.js.